python/pyxet/pyxet/file_system.py

```diff
@@ -578,6 +578,35 @@ def mv(self, path1, path2, *args, **kwargs):
     def move(self, path1, path2, *args, **kwargs):
         return self.mv(path1, path2, *args, **kwargs)
 
+    def add_deduplication_hints(self, path_urls, min_dedup_byte_threshhold = None):
+        """
+        Fetches and downloads all of the metadata needed for binary deduplication against 
+        all the paths given by `paths`.  Once fetched, new data will be deduplicated against 
+        any binary content given by `paths`.  
+        """
+        
+        url_paths = [parse_url(url, self.domain, partial_remote = True) for url in path_urls]
+
+        self._add_deduplication_hints_by_url(url_paths, min_dedup_byte_threshhold)
+    
+
+    def _add_deduplication_hints_by_url(self, url_paths, min_dedup_byte_threshhold = None):
+
+        if min_dedup_byte_threshhold is None:
+            # TODO: set this to a more resonable value.  
+            min_dedup_byte_threshhold = 0 
+
+        paths_by_remotes = {}
+        for url in url_paths:
+             paths_by_remotes.setdefault(url.remote, []).append(url)
+
+        for (remote, urls) in paths_by_remotes.items():
+            repo_handle = _manager.get_repo(remote)
```

Comment:
We should be very careful here that hint shards are not from other repos because they don't share shards.

Reply:
The shards are downloaded according to the repo, so the line below will only retrieve shards of that repo and store it with the local snapshot, so this won't be a problem.